sunkit_image/radial.py

```diff
@@ -577,3 +583,111 @@ def fnrgf(
+    """
+    Implementation of the Radial Histogram Equalizing Filter (RHEF).
+
+    The filter works as follows:
+
+    Radial Histogram Equalization is a simple algorithm for removing the radial gradient to reveal
+    coronal structure. It also significantly improves the visualization of high dynamic range solar imagery.
+    It takes the input map and bins the pixels by radius, then ranks each element sequentially and normalizes the pixels at that radius.
+
+    .. note::
+
+        After applying the filter, the maps need to be plotted with smap.plot(norm=False) to be visualized correctly.
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        The sunpy map to enhance.
+    radial_bin_edges : `astropy.units.Quantity`
+        A two-dimensional array of bin edges of size ``[2, nbins]`` where ``nbins`` is
+        the number of bins.
+    application_radius : `astropy.units.Quantity`, optional
+        The RHEF is applied to emission at radii above the application_radius.
+        Defaults to 0 solar radii.
+    upsilon : None or int or tuple of size 2, optional
+        A double-sided gamma function applied to the equalized histograms.
+        See Equation (4.15) in the thesis
+        Defaults to 0.35
+    method : str
+        A string describing which method to use for sorting.
+        Options are {"inplace", "numpy", "scipy"}.
+        Defaults to "numpy".
+    vignette : None or Bool or `astropy.units.Quantity`, optional
+        Set pixels above this radius to black.
+        Defaults to None.
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A SunPy map that has had the RHEF applied to it.
+
+    References
+    ----------
+    * Gilly & Cranmer 2024, in prep.
+
+    * The implementation is highly inspired by this doctoral thesis:
+      Gilly, G. Spectroscopic Analysis and Image Processing of the Optically-Thin Solar Corona
+      https://www.proquest.com/docview/2759080511
+    """
+
+    # Get the radii for every pixel
+    map_r = find_pixel_radii(smap).to(u.R_sun)
+
+    if radial_bin_edges is None:
+        radial_bin_edges = equally_spaced_bins(0, 2, smap.data.shape[0] // 2)
+        radial_bin_edges *= u.R_sun
+
+    # Make sure bins are in the map.
+    if radial_bin_edges[1, -1] > np.max(map_r):
+        radial_bin_edges = equally_spaced_bins(
+            inner_value=radial_bin_edges[0, 0],
+            outer_value=np.max(map_r),
+            nbins=radial_bin_edges.shape[1],
+        )
+
+    # Select the sort method
+    if method == "inplace":
+        the_func = percentile_ranks_numpy_inplace
+    elif method == "numpy":
+        the_func = percentile_ranks_numpy
+    elif method == "scipy":
+        the_func = percentile_ranks_scipy
+    else:
+        msg = f"{method} is invalid. Allowed values are 'inplace', 'numpy', 'scipy'"
+        raise NotImplementedError(msg)
+
+    # Allocate storage for the filtered data
+    data = np.zeros_like(smap.data)
+    meta = smap.meta
+    disable_pb = True
```

Comment:
This turns off the tqdm for this?

Reply:
I believe that it completely disables the progressbar. This could probably be a parameter in the function, because sometimes you want to watch a slow thing finish and other times you want to do 100 things quickly

Reply:
Yeah, we should add a way to disable it for a user.